src/librustc/ty/adjustment.rs

```diff
@@ -13,10 +13,83 @@ use hir::def_id::DefId;
 use ty::{self, Ty, TyCtxt, TypeAndMut};
 use ty::subst::Substs;
 
+
+/// Represents coercing a value to a different type of value.
+///
+/// We transform values by following the following steps in order:
+/// 1. Apply a step of `Adjust` (see its variants for details).
```

Comment:
What do you mean by "base"? But yeah I could use `kind` and then refer to "variants of `Adjust`".